### PR TITLE
Add support for running tests on .NET Core

### PR DESCRIPTION
**Pull request by dsplaisted**
This PR lets us run xUnit tests on .NET Core.  It supports F5 debugging, so you can just set a test project as the startup project and start debugging to run the tests.  (However, it currently won't pause at the end of the tests, so unless you hit a breakpoint or something similar, the output from the tests will disappear before you can see the results.)

**NOTE:** This PR depends on pull request dotnet/buildtools#276, which adds a .props file to the xunit.console.netcore package to deploy the .exe to the output folder.

The MSBuild.NetCoreDependencies project is used to deploy managed .exe files (such as the test runner or C#/VB compilers) to the output folder when building for .NET Core.  I tried a variety of ways to deploy the xunit console runner to the output folder, and this is what I settled on.  Feedback is welcome if you have any better ideas.

NuGet v3 ignores .exe files in lib folders.  The mechanism it provides to deploy arbitrary files to the output folder is to include an MSBuild .props file in the package under build/framework_identifier, which includes the file as an item to be copied to the output folder.  However, this does not work when there are multiple frameworks defined in project.json, as is the case for the MSBuild projects (since there are different build configurations to switch between building for .NET Core or .NET Framework).

NuGet v3 also supports deploying arbitrary files via "native" folders in a "runtimes" folder in the package.  However, this would require the files to be
specific to a given runtime, or at least for the same files to be duplicated in the package for all possible runtimes.

To work around this for MSBuild, the MSBuild.NetCoreDependencies project is built only for .NET Core.  Other projects which need the .exe files to be
deployed can add a project reference to this project, conditional on the `NetCoreBuild` property being true.

### CODE REVIEW

**Comment by AndyGerlicher** [comment]
LGTM

File: src/MSBuild.NetCoreDependencies/MSBuild.NetCoreDependencies.nuget.props [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup Condition="'$(NuGetPackageRoot)' == ''">
+    <NuGetPackageRoot>C:\Users\daplaist\.nuget\packages\</NuGetPackageRoot>
```
Comment (akoeplinger) [inline src/MSBuild.NetCoreDependencies/MSBuild.NetCoreDependencies.nuget.props]: Do we need NuGetPackageRoot set or can this be removed?

Comment (rainersigwald) [inline src/MSBuild.NetCoreDependencies/MSBuild.NetCoreDependencies.nuget.props]: In other projects, we didn't check in this file.  I thought I added it to `.gitignore`, actually.  Do we need to check it in here?

File: src/dir.targets [hunk 2]
```diff
@@ -1,6 +1,10 @@
 ﻿<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\dir.targets" />
 
+  <PropertyGroup>
+    <CopyNuGetImplementations Condition="'$(IsTestProject)' == 'true'">true</CopyNuGetImplementations>
```
Comment (rainersigwald) [inline src/dir.targets]: :+1:

**Comment by rainersigwald** [comment]
LGTM if we can delete or justify the `nuget.props` file.

File: src/MSBuild.NetCoreDependencies/MSBuild.NetCoreDependencies.nuget.props (hunk 1 above)
Comment (dsplaisted) [inline src/MSBuild.NetCoreDependencies/MSBuild.NetCoreDependencies.nuget.props]: We had an ignore for .nuget.targets but not .nuget.props.  I've added the .props ignore and removed the file from the repo.

**Comment by dsplaisted** [comment]
I've updated this PR based on the changes which have been merged in dotnet/buildtools#276.  This is ready to merge if it looks good.

**Comment by rainersigwald** [comment]
LGTM.  When you've merged I'll pick up the rest of the test assemblies.

**Pull request merged**